forc-plugins/forc-client/src/op/call/mod.rs

```diff
@@ -0,0 +1,713 @@
+mod missing_contracts;
+mod parser;
+
+use crate::{
+    cmd,
+    constants::DEFAULT_PRIVATE_KEY,
+    op::call::{
+        missing_contracts::get_missing_contracts,
+        parser::{param_type_val_to_token, token_to_string},
+    },
+    util::{
+        node_url::get_node_url,
+        tx::{prompt_forc_wallet_password, select_local_wallet_account},
+    },
+};
+use anyhow::{bail, Result};
+use either::Either;
+use fuel_abi_types::abi::unified_program::UnifiedProgramABI;
+use fuels::{
+    accounts::{provider::Provider, wallet::WalletUnlocked},
+    crypto::SecretKey,
+    programs::calls::{
+        receipt_parser::ReceiptParser,
+        traits::{ContractDependencyConfigurator, TransactionTuner},
+        ContractCall,
+    },
+};
+use fuels_core::{
+    codec::{
+        encode_fn_selector, log_formatters_lookup, ABIDecoder, ABIEncoder, DecoderConfig,
+        EncoderConfig, LogDecoder,
+    },
+    types::{
+        bech32::Bech32ContractId,
+        param_types::ParamType,
+        transaction::{Transaction, TxPolicies},
+        transaction_builders::{BuildableTransaction, ScriptBuildStrategy, VariableOutputPolicy},
+        ContractId,
+    },
+};
+use std::{collections::HashMap, str::FromStr};
+
+/// A command for calling a contract function.
+pub async fn call(cmd: cmd::Call) -> anyhow::Result<String> {
+    let cmd::Call {
+        contract_id,
+        abi,
+        function,
+        args,
+        node,
+        caller,
+        call_parameters,
+        mode,
+        gas,
+        external_contracts,
+        output,
+    } = cmd;
+    let node_url = get_node_url(&node, &None)?;
+    let provider: Provider = Provider::connect(node_url).await?;
+
+    let wallet = get_wallet(caller, provider).await?;
+
+    let cmd::call::FuncType::Selector(selector) = function;
+    let abi_str = match abi {
```

Comment:
would also be cool to try and fetch the verified ABI from the registry if only the contract id was given with no ABI file... obviously this isn't quite ready yet but would be a nice addition. Perhaps leaving a comment or linking to an issue to integrate this would be good.

Reply:
https://github.com/FuelLabs/sway/pull/6791/commits/46e5007e434824916b964b26de81e6714e008047